libraries/botbuilder-ai/tests/qnaMakerRecognizer.test.js

```diff
@@ -27,76 +33,159 @@ const createMessageActivity = (text) => {
+        strictEqual(result.answers, undefined);
+        strictEqual(result.intents.QnAMatch, undefined);
+        notStrictEqual(result.intents.None, undefined);
     });
 
     it('Return Answers', async function () {
         nock(hostname)
             .post(/knowledgebases/)
             .replyWithFile(200, testDataFolder + 'QnaMaker_ReturnsAnswer.json');
-        const activity = createMessageActivity('test');
-        const dc = new DialogContext(new DialogSet(), new TurnContext(new TestAdapter(), activity), {});
+        const activity = createMessageActivity(qnaIntentText);
+        const dc = createContext(activity);
         const result = await recognizer.recognize(dc, activity);
         validateAnswers(result);
-        assert.equal(result.intents.None, undefined);
-        assert.notEqual(result.intents.QnAMatch, undefined);
+        strictEqual(result.intents.None, undefined);
+        notStrictEqual(result.intents.QnAMatch, undefined);
     });
 
     it('Top N Answers', async function () {
         nock(hostname)
             .post(/knowledgebases/)
             .replyWithFile(200, testDataFolder + 'QnaMaker_TopNAnswer.json');
-        const activity = createMessageActivity('test');
-        const dc = new DialogContext(new DialogSet(), new TurnContext(new TestAdapter(), activity), {});
+        const activity = createMessageActivity(qnaIntentText);
+        const dc = createContext(activity);
         const result = await recognizer.recognize(dc, activity);
         validateAnswers(result);
-        assert.equal(result.intents.None, undefined);
-        assert.notEqual(result.intents.QnAMatch, undefined);
+        strictEqual(result.intents.None, undefined);
+        notStrictEqual(result.intents.QnAMatch, undefined);
     });
 
     it('Return Answers with Intents', async function () {
         nock(hostname)
             .post(/knowledgebases/)
             .replyWithFile(200, testDataFolder + 'QnaMaker_ReturnsAnswerWithIntent.json');
-        const activity = createMessageActivity('test');
-        const dc = new DialogContext(new DialogSet(), new TurnContext(new TestAdapter(), activity), {});
+        const activity = createMessageActivity(qnaIntentText);
+        const dc = createContext(activity);
         const result = await recognizer.recognize(dc, activity);
         validateAnswers(result);
-        assert.equal(result.intents.None, undefined);
-        assert.notEqual(result.intents.DeferToRecognizer_xxx, undefined);
+        strictEqual(result.intents.None, undefined);
+        notStrictEqual(result.intents.DeferToRecognizer_xxx, undefined);
+    });
+
+    describe('Telemetry', () => {
+        let spy;
+
+        beforeEach(() => {
+            spy = spyOnTelemetryClientTrackEvent(recognizer);
+        });
+
+        afterEach(() => {
+            spy.restore();
+        });
+
+        it('logs PII when logPersonalInformation is true', async () => {
+            nock(hostname)
```

Comment:
Would be good to add the requisite nock `disableNetConnect`/`cleanAll` calls as well.